### PR TITLE
feat: add block explorer links on swap feedback and wallet info

### LINKED ISSUE #307: Add block-explorer links on transactions
1. Add the transaction hash with a link to the transaction's page on the block explorer
2. When clicking the wallet address, go to the address's block explorer page to view all its transactions
3. Make sure the block explorer works with the current `devnet` URL, etc.

### PR DESCRIPTION
**Pull request by pedronauck**
✨ Fix #307

### CODE REVIEW

**Review by adlerjohn — COMMENTED** [review]
Tried it out and it looks pretty good! The block explorer isn't showing anything, but that's the block explorer's problem. 👍

**Comment by luizstacio** [comment]
@pedronauck looks like we are getting the wrong tx as id, this is why block-explorer is not showing the correct tx: 
https://fuellabs.github.io/block-explorer-v2/transaction/0x13d349c50bed0d5449b55755db28e5b9b0dd8d2ce4488ea91efd04b311fe077d

**Comment by adlerjohn** [comment]
> this is why block-explorer is not showing the correct tx

Are you sure? I was under the impression the block explorer was connected to the Dubai testnet, which isn't live anymore.

**Comment by pedronauck** [comment]
I'm getting the `data.blockId` returned from [TransactionResult](https://github.com/FuelLabs/fuels-ts/blob/dc12dfb9d46c0a708cfdf8ff86809c697a40856d/packages/providers/src/transaction-response/transaction-response.ts#L48)

I checked and there's no other id returned from it @luizstacio

File: packages/app/src/systems/Core/components/WalletInfo.tsx [hunk 1]
```diff
@@ -35,6 +37,15 @@ export function WalletInfo({ onClose }: WalletInfoProps) {
           <AssetItem key={coin.assetId} coin={coin} />
         </div>
       ))}
+      <div className="flex justify-center mt-4 text-sm">
+        <Link
+          isExternal
+          className="flex items-center gap-2"
+          href={`${BLOCK_EXPLORER_URL}/address/${wallet?.address}`}
```
Comment (luizstacio) [inline packages/app/src/systems/Core/components/WalletInfo.tsx:44]: Maybe add already the providerUrl as query param.

`${BLOCK_EXPLORER_URL}/address/${wallet?.address}?provider_url=${VITE_FUEL_PROVIDER_URL}`

**Review by luizstacio — CHANGES_REQUESTED** [review]

File: packages/app/src/systems/Core/components/WalletInfo.tsx (hunk 1 above)
Comment (pedronauck) [reply packages/app/src/systems/Core/components/WalletInfo.tsx]: ✅

File: packages/app/src/systems/Mint/hooks/useMint.ts [hunk 2]
```diff
@@ -21,18 +21,17 @@ export function useMint(opts: UseMintOpts = {}) {
   const mutation = useMutation(
     async () => {
       if (!txCost.total) return;
-      await methods.mint(txCost.total);
+      return methods.mint(txCost.gas);
```
[the remark this answers is not in the document]
Comment (pedronauck) [reply packages/app/src/systems/Mint/hooks/useMint.ts]: ✅

File: packages/app/src/systems/Swap/utils/queries.ts [hunk 3]
```diff
@@ -69,48 +67,47 @@ export const swapTokens = async (
...
-    await contract.submit.swap_with_minimum(
+
+    return contract.submitResult.swap_with_minimum(
       minValue.amount,
       DEADLINE,
       getOverrides({
         forward: [amount, coinFrom.assetId],
         gasLimit: txCost.total,
-        variableOutputs: 1,
+        variableOutputs: 2,
```
[the remark this answers is not in the document]
Comment (pedronauck) [reply packages/app/src/systems/Swap/utils/queries.ts:92]: ✅

**Review by luizstacio — APPROVED** [review]

**Pull request merged**